devlib/utils/types.py

```diff
@@ -68,6 +68,15 @@ def numeric(value):
     """
     if isinstance(value, int):
```

Comment:
In short, because  ℤ ⊆ ℝ, and we want to provide as "tight" a value a possible.

There are certain values that are conceptually always ints (e.g. counts), and there are analysis tools that may expect them to be such. On the other hand, there are applications which, either for simplicity or out of necessity (e.g. they're implemented in Javascript), report all numeric values as floats. Things then break when you try to combine the two.

It's also partially a usability thing, e.g. `pandas` will format ints in a nicer, more readable way.

Basically, if something expects a float and we give it an int, chances are we're not losing anything; if something expects an int and we give it a float, things break. Hence, as the default behavior,  prefer ints.